### PR TITLE
KAZOO-5260_4oh: fix number release of unknown carrier

### PR DESCRIPTION
**Pull request by fenollp**
Note: the catch-all was added so that binary_to_existing_atom failure is caught.

### CODE REVIEW

File: core/kazoo_number_manager/src/carriers/knm_carriers.erl [hunk 1]
```diff
@@ -325,7 +325,7 @@ disconnect(Number) ->
     try apply(Module, disconnect_number, [Number]) of
         Result -> Result
     catch
-        'error':'undef' ->
+        'error':_ ->
```
Comment (lazedo) [inline core/kazoo_number_manager/src/carriers/knm_carriers.erl]: @fenollp why this change ? what other values are you seeing here ?

Comment (fenollp) [inline core/kazoo_number_manager/src/carriers/knm_carriers.erl]: > Note: the catch-all was added so that binary_to_existing_atom failure is caught.

`binary_to_existing_atom` fails with `badarg`.
